src/ops/constant.rs

```diff
@@ -229,3 +235,71 @@ pub trait CustomConst:
 
 impl_downcast!(CustomConst);
 impl_box_clone!(CustomConst, CustomConstBoxClone);
+
+#[cfg(test)]
+mod test {
+    use super::ConstValue;
+    use crate::{
+        builder::{BuildError, Container, DFGBuilder, Dataflow, DataflowHugr},
+        type_row,
+        types::{ClassicType, SimpleType, TypeRow},
+    };
+
+    #[test]
+    fn test_predicate() -> Result<(), BuildError> {
+        let pred_rows = vec![
+            type_row![
+                SimpleType::Classic(ClassicType::i64()),
+                SimpleType::Classic(ClassicType::F64)
+            ],
+            type_row![],
+        ];
+        let pred_ty = SimpleType::new_predicate(pred_rows.clone());
+
+        let mut b = DFGBuilder::new(type_row![], TypeRow::from(vec![pred_ty.clone()]))?;
+        let c = b.add_constant(ConstValue::predicate(
+            0,
+            ConstValue::Tuple(vec![ConstValue::i64(3), ConstValue::F64(3.15)]),
+            pred_rows.clone(),
+        ))?;
+        let w = b.load_const(&c)?;
+        b.finish_hugr_with_outputs([w]).unwrap();
+
+        let mut b = DFGBuilder::new(type_row![], TypeRow::from(vec![pred_ty]))?;
+        let c = b.add_constant(ConstValue::predicate(
+            1,
+            ConstValue::Tuple(vec![]),
+            pred_rows,
+        ))?;
+        let w = b.load_const(&c)?;
+        b.finish_hugr_with_outputs([w]).unwrap();
+
+        Ok(())
+    }
+
+    #[test]
+    #[should_panic] // Pending resolution of https://github.com/CQCL-DEV/hugr/issues/231
+    fn test_bad_predicate() {
+        let pred_rows = vec![
+            type_row![
+                SimpleType::Classic(ClassicType::i64()),
+                SimpleType::Classic(ClassicType::F64)
+            ],
+            type_row![],
+        ];
+        let pred_ty = SimpleType::new_predicate(pred_rows.clone());
+
+        let mut b = DFGBuilder::new(type_row![], TypeRow::from(vec![pred_ty])).unwrap();
+        // Until #231 is fixed, this is made to fail by an assert in ConstValue::predicate
+        let c = b
+            .add_constant(ConstValue::predicate(
+                0,
+                ConstValue::Tuple(vec![]),
+                pred_rows,
+            ))
+            .unwrap();
+        let w = b.load_const(&c).unwrap();
+        // When #231 is fixed, there should be a validation error here instead
+        b.finish_hugr_with_outputs([w]).unwrap_err();
```

Comment:
I think it would be better to instead match on the correct error here, since then the fix to #231 should make this test fail, forcing it to be updated correctly (i.e. the `#[should_panic]` being removed)

Reply:
I'll claim the excuse that until #258 I wasn't sure what the correct error *was*....but now I know ;-). Good call, done.